platforms/core-configuration/base-diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DiagnosticsTaskNames.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.plugins.HelpTasksPlugin;
+
+/**
+ * The names of the diagnostic tasks added by the {@link HelpTasksPlugin} and the
+ * {@code SoftwareReportingTasksPlugin}.
+ *
+ * @since 8.13
+ */
+@Incubating
+public interface DiagnosticsTaskNames {
```

Comment:
I wonder if this belongs as public API. 

We already have these names exposed on the HelpTasksPlugin itself, and I think it might be confusing having them exposed in multiple places.

If we expose them here than I would vote to deprecate the static properties on the HelpTasksPlugin

Reply:
See above.